llvm/lib/Transforms/Instrumentation/SoftPointerAuth.cpp

```diff
@@ -825,12 +835,15 @@ bool SoftPointerAuth::transformPointerAuthCall(CallSite oldCall,
   CallSite newCall;
   if (oldCall.isInvoke()) {
     auto oldInvoke = cast<InvokeInst>(oldCall.getInstruction());
-    newCall = builder.CreateInvoke(callee,
-                                   oldInvoke->getNormalDest(),
-                                   oldInvoke->getUnwindDest(),
-                                   args, opBundles);
+    newCall = builder.CreateInvoke(
+        cast<FunctionType>(
+            cast<PointerType>(callee->getType())->getElementType()),
+        callee, oldInvoke->getNormalDest(), oldInvoke->getUnwindDest(), args,
+        opBundles);
   } else {
-    newCall = builder.CreateCall(callee, args, opBundles);
+    newCall = builder.CreateCall(
+        cast<FunctionType>(callee->getType()->getPointerElementType()), callee,
+        args, opBundles);
```

Comment:
You should pull the callee type off the old call site in these cases.

Reply:
Done.